### PR TITLE
Add hidden `--no-implicit-pubspec-resolution` flag for one stable release.

### LINKED ISSUE #157532: Add `--no-emit-legacy-flutter-plugins` (which does not write `.flutter-plugins`)
Was optimistically documented in https://github.com/flutter/website/pull/11303/files.

I'd like to give our users a stable release where legacy plugins warn (see https://github.com/flutter/flutter/pull/157388, thanks @reidbaker for the suggestion), but continue to work. As part of that, I'd like to ship a hidden flag, `--no-emit-legacy-flutter-plugins`, that does not create a `.flutter-plugins` file to let end-users test their workflows/run on CI/etc.

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/flutter/flutter/issues/157532.

Work towards https://github.com/flutter/flutter/issues/157819.

Because this flag is checked in `FlutterCommand.verifyAndRun`, it seemed cleaner to just make it a global flag - particularly because this flag's longevity is unlikely to be longer than a single stable release.

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/permeable/packages_test.dart [hunk 1]
```diff
@@ -610,7 +623,37 @@ workspace:
...
+        fileSystem: globals.fs,
+        logger: globals.logger,
+        processManager: globals.processManager,
+        usage: globals.flutterUsage,
+        botDetector: globals.botDetector,
+        platform: globals.platform,
+        stdio: mockStdio,
+      ),
+    });
+
```
Comment (reidbaker) [inline packages/flutter_tools/test/commands.shard/permeable/packages_test.dart:639]: nit extra whitespace

**Review by reidbaker — APPROVED** [review]

**Comment by reidbaker** [comment]
You might choose to leave the definition of this flag around for another release where the flag does nothing since this flag is an opt in to the new behavior and I think undefined flags error. We don't need to decide that now and there are also good arguments for code health.

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 2]
```diff
@@ -124,6 +125,23 @@ class FlutterCommandRunner extends CommandRunner<void> {
       help: 'Print the address of the Dart Tooling Daemon, if one is hosted by the Flutter CLI.',
       hide: !verboseHelp,
     );
+
+    // TODO(matanlurey): Remove after the Q4 2024 stable release; this is intended
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart:129]: probably Q1 2025 since Q4 has already cut

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/runner/flutter_command_runner.dart:129]: or maybe Q2 2025?

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart (hunk 2 above)
Comment (matanlurey) [reply packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: Done!

**Comment by matanlurey** [comment]
@jonahwilliams has an interesting idea on renaming and reusing this flag as a bundle of features versus a 1-off.

Will post a new GH issue and go from there.

**Comment by matanlurey** [comment]
I am merging this after renaming the flag `--implicit-pubspec-resolution` as discussed (https://github.com/flutter/flutter/issues/157819).

**Pull request merged**